NBitcoin/PubKey.cs

```diff
@@ -465,6 +465,33 @@ public PubKey GetSharedPubkey(Key key)
 			return new ECKey(pubkey.GetEncoded(true), false).GetPubKey(true);
 		}
 
+		public string Encrypt(string message)
+		{
+			if(string.IsNullOrEmpty(message))
+				throw new ArgumentNullException(nameof(message));
+
+			var bytes = Encoding.UTF8.GetBytes(message);
+			return Encoders.Base64.EncodeData(Encrypt(bytes));
+		}
+
+		public byte[] Encrypt(byte[] message)
+		{
+			if(message is null)

```

Comment:
it's the first time I see a null check using `is` instead of `==`, does this work?

Reply:
Actually I read good reasons to use it and why it is better and more performant in some cases.
I think it is because it does not have to check with overload operators or stuff like this... trying to find the article.

Reply:
yes. https://stackoverflow.com/questions/53184991/c-sharp-how-to-check-for-null-value-is-null-or-null-value-can-we-use-is

This PR is ready for review @NicolasDorier @nopara73. I think more tests are not necessary and I verified it is compatible with Electrum wallet.

Reply:
Actually now I am seeing article which say it is slower. (not that it really matter in this specific case though)

Reply:
@NicolasDorier slower? lontivero's link says the opposite